File: src/main/kotlin/com/miumiuhaskeer/fastmessage/statistic/model/entity/Role.java
package com.miumiuhaskeer.fastmessage.statistic.model.entity;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;

/**
 * Class copied and modified from FastMessage project
 */
@Entity
@Getter
@Setter
@Table(name = "fm_role")
public class Role {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "id_gen")
    @SequenceGenerator(name = "id_gen", sequenceName = "fm_role_id_seq", allocationSize = 1)
    private Long id;

    @Column(length = 50)
    @Enumerated(EnumType.STRING)
    private ERole name;

    public Role() {
    }

    public Role(ERole name) {
        this.name = name;
    }
}

File: src/main/kotlin/com/miumiuhaskeer/fastmessage/statistic/model/entity/Role.kt
package com.miumiuhaskeer.fastmessage.statistic.model.entity

import javax.persistence.*

@Entity
@Table(name = "fm_role")
class Role {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "id_gen")
    @SequenceGenerator(name = "id_gen", sequenceName = "fm_role_id_seq", allocationSize = 1)
    var id: Long? = null

    @Column(length = 50)
    @Enumerated(EnumType.STRING)
    var name: ERole? = null
}